docs/adr/adr-004-state-interaction.md

```diff
@@ -57,6 +57,12 @@ type StateAccessor interface {
 3. **LOCAL**: eventually, **full** nodes will be able to provide state to the network, and can therefore execute and 
 respond to the queries without relaying queries to celestia-core (*to be scoped out and implemented in later iterations*)
 
+### Verification of Balances
+In order to check that the balances returned via the `AccountBalance` query are correct, it is necessary to also request
+Merkle proofs from the celestia-app and verify them against the latest head's `AppHash`. In order for the `StateAccessor`
+to do this, it would need access to the `header.Store`'s `Head()` method in order to get the latest known header of the 
+node and check its `AppHash`.
```

Comment:
@renaynay, this is doesn't take into account the syncing aspect. During the sync, we should either (1)block until syncing is done or (2)we should rely on the latest header in the pending hash. Note that headers coming to the pending hash are verified non-adjacently following Tendermint consensus rules with code adapted from Tendermint software. Thus, we can trust headers in the pending cache.

That's being said, using only `store.Head` is not reliable for cases when a node is syncing, e.g. you just opened the lid of the laptop and want to send a tx.

Reply:
`StateService` should not be started / made available on the node while node is syncing, along with many other methods in the Public API you've listed so far. The question is, do we pass sync state to every module that relies on being synced to be able to either serve requests / operate? Or should this be done via some sort of service manager on the node that waits for header service to sync and then kicks off the other dependent services? 

WDYT?

Reply:
Decision from @Wondertan: 

Let's have services rely on syncer.NetworkHead() and services can be made available and work off network head instead of waiting to be synced completely

Reply:
Please review [dc0a0ad](https://github.com/celestiaorg/celestia-node/pull/513/commits/dc0a0ad3ea9728fff0473078de57d2b1b0b74bb0) @Wondertan